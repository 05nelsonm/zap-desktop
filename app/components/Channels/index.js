export ChannelCapacity from './ChannelCapacity'
export ChannelMoreButton from './ChannelMoreButton'
export ChannelStatus from './ChannelStatus'
export ChannelSummaryList from './ChannelSummaryList'
export ChannelSummaryListItem from './ChannelSummaryListItem'
