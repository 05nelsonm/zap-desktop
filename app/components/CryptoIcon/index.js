import CryptoIcon from './CryptoIcon'

export default CryptoIcon
