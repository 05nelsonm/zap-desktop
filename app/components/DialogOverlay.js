import styled from 'styled-components'
import { Flex } from 'rebass'

const Overlay = styled(Flex)`
  position: absolute;
  z-index: 99999;
  background-color: rgba(255, 255, 255, 0.25);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

export default Overlay
