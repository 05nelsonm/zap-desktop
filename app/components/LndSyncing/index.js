import LndSyncing from './LndSyncing'

export default LndSyncing
