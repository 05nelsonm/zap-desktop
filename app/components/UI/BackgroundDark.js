import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundDark
 * @example
 * <BackgroundDark />
 */
class BackgroundDark extends React.Component {
  render() {
    return <Box bg="darkestBackground" color="primaryText" {...this.props} />
  }
}

export default BackgroundDark
