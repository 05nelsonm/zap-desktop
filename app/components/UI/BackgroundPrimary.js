import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundPrimary
 * @example
 * <BackgroundPrimary />
 */
class BackgroundPrimary extends React.Component {
  render() {
    return <Box bg="primaryColor" color="primaryText" {...this.props} />
  }
}

export default BackgroundPrimary
