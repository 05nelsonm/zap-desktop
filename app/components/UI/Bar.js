import styled from 'styled-components'
import { variant, space, height } from 'styled-system'

const bars = variant({
  key: 'bars',
})

const Bar = styled.hr`
  height: 1px;
  margin: 0;
  padding: 0;
  border: 0;
  background-color: ${props => props.theme.colors.primaryText};
  ${space}
  ${height}
  ${bars}
`

Bar.defaultProps = {
  variant: 'normal',
}

Bar.displayName = 'Bar'

export default Bar
