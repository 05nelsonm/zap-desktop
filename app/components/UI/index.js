export ActionBar from './ActionBar'
export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export Button from './Button'
export Checkbox from './Checkbox'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export Notification from './Notification'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Value from './Value'
export Wizard from './Wizard'
