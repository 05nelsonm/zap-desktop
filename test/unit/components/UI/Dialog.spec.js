import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Flex } from 'rebass'
import { Dialog, Text, Heading, Button } from 'components/UI'
import Globe from 'components/Icon/Globe'

describe('component.UI.Dialog', () => {
  it('should render correctly with two buttons', () => {
    const wrapper = shallow(
      <Dialog
        header="Title"
        onClose={() => {}}
        buttons={[{ name: 'Delete', onClick: () => {} }, { name: 'Cancel', onClick: () => {} }]}
      />
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render correctly with one button', () => {
    const wrapper = shallow(
      <Dialog caption="Title" onClose={() => {}} buttons={[{ name: 'Ok', onClick: () => {} }]} />
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render correctly with custom content', () => {
    const CustomDialog = () => {
      const customHeader = (
        <Flex flexDirection="column" alignItems="center" mb={3} alignSelf="stretch">
          <Globe width={64} height={64} color="white" />
          <Heading.h2>Custom heading</Heading.h2>
        </Flex>
      )

      const customButtons = (
        <>
          <Button key="0" variant="primary">
            Primary
          </Button>
          <Button key="1" variant="secondary">
            Secondary
          </Button>
        </>
      )

      return (
        <Dialog header={customHeader} onClose={() => alert('closed')} buttons={customButtons}>
          <Flex flexDirection="column" alignItems="center">
            <Text fontSize="xl">This is a custom body</Text>
          </Flex>
        </Dialog>
      )
    }

    const wrapper = shallow(<CustomDialog />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
