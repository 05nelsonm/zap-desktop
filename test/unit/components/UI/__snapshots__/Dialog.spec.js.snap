// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Dialog should render correctly with custom content 1`] = `
<Dialog
  buttons={
    <React.Fragment>
      <ForwardRef
        active={false}
        processing={false}
        size="medium"
        variant="primary"
      >
        Primary
      </ForwardRef>
      <ForwardRef
        active={false}
        processing={false}
        size="medium"
        variant="secondary"
      >
        Secondary
      </ForwardRef>
    </React.Fragment>
  }
  header={
    <ForwardRef
      alignItems="center"
      alignSelf="stretch"
      flexDirection="column"
      mb={3}
    >
      <SvgGlobe
        color="white"
        height={64}
        width={64}
      />
      <Heading2>
        Custom heading
      </Heading2>
    </ForwardRef>
  }
  onClose={[Function]}
  width={550}
>
  <Styled(styled.div)
    alignItems="center"
    flexDirection="column"
  >
    <Text
      fontSize="xl"
    >
      This is a custom body
    </Text>
  </Styled(styled.div)>
</Dialog>
`;

exports[`component.UI.Dialog should render correctly with one button 1`] = `
<Styled(styled.div)
  bg="primaryColor"
  borderRadius={5}
  boxShadow="0 3px 4px 0 rgba(0, 0, 0, 0.5)"
  width={550}
>
  <CloseButton
    onClick={[Function]}
  />
  <Styled(styled.div)
    alignItems="center"
    alignSelf="stretch"
    flexDirection="column"
    justifyContent="center"
    pb={4}
  >
    <Styled(styled.div)
      alignItems="flex-start"
      flexDirection="column"
      justifyContent="center"
    >
      <Heading2
        mb={4}
      >
        Title
      </Heading2>
    </Styled(styled.div)>
    <Styled(styled.div)
      alignSelf="stretch"
      flexDirection="row"
      justifyContent="space-evenly"
      mt={4}
      pl={6}
      pr={6}
    >
      <Button
        active={false}
        key="Ok"
        onClick={[Function]}
        processing={false}
        size="medium"
        variant="primary"
      >
        Ok
      </Button>
    </Styled(styled.div)>
  </Styled(styled.div)>
</Styled(styled.div)>
`;

exports[`component.UI.Dialog should render correctly with two buttons 1`] = `
<Styled(styled.div)
  bg="primaryColor"
  borderRadius={5}
  boxShadow="0 3px 4px 0 rgba(0, 0, 0, 0.5)"
  width={550}
>
  <CloseButton
    onClick={[Function]}
  />
  <Styled(styled.div)
    alignItems="center"
    alignSelf="stretch"
    flexDirection="column"
    justifyContent="center"
    pb={4}
  >
    <Styled(styled.div)
      alignItems="flex-start"
      flexDirection="column"
      justifyContent="center"
    >
      <Heading2
        mb={4}
      >
        Title
      </Heading2>
    </Styled(styled.div)>
    <Styled(styled.div)
      alignSelf="stretch"
      flexDirection="row"
      justifyContent="space-evenly"
      mt={4}
      pl={6}
      pr={6}
    >
      <Button
        active={false}
        key="Delete"
        onClick={[Function]}
        processing={false}
        size="medium"
        variant="primary"
      >
        Delete
      </Button>
      <Button
        active={false}
        key="Cancel"
        onClick={[Function]}
        processing={false}
        size="medium"
        variant="normal"
      >
        Cancel
      </Button>
    </Styled(styled.div)>
  </Styled(styled.div)>
</Styled(styled.div)>
`;
